Author: Danny Milosavljevic <dannym@friendly-machines.com>
Date: 2026-05-02
Subject: Use local AllFilesMask constant
License: expat

FPC 2.6.0 uses version-number {$if ...} expressions in cfileutl.pas to
choose between an RTL-provided AllFilesMask and a local fallback.  The
bootstrap translator does not need the RTL-provided constant here.

Always define AllFilesMask locally, and remove the version-number
conditionals so the bootstrap translator does not need to evaluate
FPC_VERSION/FPC_RELEASE/FPC_PATCH in {$if} expressions.

--- a/compiler/cfileutl.pas
+++ b/compiler/cfileutl.pas
@@ -147,18 +147,6 @@
 
 {$undef AllFilesMaskIsInRTL}
 
-{$if (FPC_VERSION > 2)}
-  {$define AllFilesMaskIsInRTL}
-{$endif FPC_VERSION}
-
-{$if (FPC_VERSION = 2) and (FPC_RELEASE > 2)}
-  {$define AllFilesMaskIsInRTL}
-{$endif}
-
-{$if (FPC_VERSION = 2) and (FPC_RELEASE = 2) and (FPC_PATCH > 0)}
-  {$define AllFilesMaskIsInRTL}
-{$endif}
-
 {$ifndef AllFilesMaskIsInRTL}
   {$if defined(go32v2) or defined(watcom)}
   const
